scripts/benchmark_ad.py

```diff
@@ -0,0 +1,130 @@
+from config.configuration import CATEGORY_TO_TYPE, DIFFUSION_AD_HPARAMS, DEFAULT_ROOT_OUTPUT_DIR
+
+
+DEFAULT_LOG_BASE = DEFAULT_ROOT_OUTPUT_DIR
+os.makedirs(DEFAULT_LOG_BASE, exist_ok=True)
+tempfile.tempdir = DEFAULT_LOG_BASE
+
+
+def parse_arguments():
+    parser = argparse.ArgumentParser(description='Anomaly Detection Benchmark Pipeline',
+                                     formatter_class=argparse.RawTextHelpFormatter,
+                                     epilog='For more information, please refer to the documentation.')
+
+    # Add arguments
+    parser.add_argument('--model', type=str, default=ModelLoader.MODEL_TO_ARG_SPECIFICS['256x256_uncond']['model_path'],
+                        help='Path to the model. Default: %(default)s')
+    parser.add_argument('--targets', type=str, nargs='+', choices=CATEGORY_TO_TYPE.keys(),
+                        help='Target categories (space-separated)')
+    parser.add_argument('--overwrite', action='store_true', default=False,
+                        help='Allow overwriting previous result for a target category')
+    parser.add_argument('--reconstruction-batch-size', type=int, default=DIFFUSION_AD_HPARAMS.reconstruction_batch_size,
+                        help='Size of the reconstruction batch. Default: %(default)s')
+
+    # Summary of the pipeline
+    pipeline_summary = '''
+    This script implements an anomaly detection benchmark pipeline. It loads a diffusion denoising model 
+    and performs automated anomaly detection evaluation. It requires specifying the target category/ies and 
+    supports optional arguments for model path, result overwriting, and reconstruction batch size.
+    '''
+
+    # Add the pipeline summary to the help screen
+    parser.description = parser.description + '\n' + pipeline_summary
+
+    args = parser.parse_args()
+
+    required_args = ['targets']  # List any required arguments here
+
+    missing_args = [arg for arg in required_args if getattr(args, arg) is None]
+    if missing_args:
+        parser.error(
+            f'The following arguments are required: {", ".join(missing_args)}')
+
+    return args
+
+
+def run_benchmark(model_path: str, target_categories: List[str], should_overwrite: bool = False):
+    # Load the model
+    loader = ModelLoader()
+    model_name = loader.get_model_name('256x256_uncond', path=model_path)
+    logger.log(f"Loading model {model_name}...")
+    model, diffusion = loader.get_model('256x256_uncond', path=model_path)
+
+    # Create the components
+    logger.log("Creating benchmark components and trainer...")
+    noiser = TimestepUniformNoiser(diffusion)
+    denoiser = ModelTimestepUniformDenoiser(model, diffusion)
+    anomaly_map_generator = BatchFilteredSquaredError()
+    anomaly_scorer = MaxValueAnomalyScorer()
+
+    diffusion_ad = DiffusionAD(
+        noiser,
+        denoiser,
+        anomaly_map_generator,
+        anomaly_scorer,
+        DIFFUSION_AD_HPARAMS,
+        model_name=model_name)
```

Comment:
extract what you can from here to another method that returns the diffusion_ad.

Reply:
like `def init_diffusion_ad()`

Reply:
Done